### PR TITLE
Sanitize whitespace in seed phrase input

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Mnemonic seed phrase input is too strict about whitespace (it fails if words have more than one space between them)

#### Summary of Changes
When "skip validation" is not enabled, sanitize whitespace between words to improve user experience

Related to https://github.com/solana-labs/solana/issues/7248

### CODE REVIEW

File: clap-utils/src/keypair.rs [hunk 1]
```diff
@@ -69,7 +69,12 @@ pub fn keypair_from_seed_phrase(
         let passphrase = prompt_passphrase(&passphrase_prompt)?;
         keypair_from_seed_phrase_and_passphrase(&seed_phrase, &passphrase)
     } else {
-        let mnemonic = Mnemonic::from_phrase(seed_phrase, Language::English)?;
+        let sanitized = seed_phrase
+            .split(' ')
+            .filter(|word| !word.is_empty())
+            .collect::<Vec<&str>>()
+            .join(" ");
```
Comment (CriesofCarrots) [inline clap-utils/src/keypair.rs]: Maybe `split_ascii_whitespace()` and `fold()`?
https://doc.rust-lang.org/std/primitive.str.html#method.split_ascii_whitespace

Comment (jstarry) [reply clap-utils/src/keypair.rs]: Sounds good, I think https://doc.rust-lang.org/std/primitive.str.html#method.split_whitespace would be better though, what do you think?

Comment (CriesofCarrots) [reply clap-utils/src/keypair.rs]: Lol, sounds good, I almost put that first

Comment (jstarry) [reply clap-utils/src/keypair.rs]: Opted to keep `join` over `fold` because `join` makes it easier to add a delimiter

**Pull request merged**